src/ntcore_c.cpp

```diff
@@ -436,3 +436,257 @@ void NT_DisposeRpcCallInfo(NT_RpcCallInfo *call_info) {
+	*last_change = value->last_change;
+	*arr_size = value->data.arr_boolean.size;
+	return value->data.arr_boolean.arr;
+}
+double *NT_GetEntryDoubleArrFromValue(struct NT_Value *value, unsigned long long *last_change, size_t *arr_size)
+{
+	if (value->type != NT_Type::NT_DOUBLE_ARRAY) return 0;
+	*last_change = value->last_change;
+	*arr_size = value->data.arr_double.size;
+	return value->data.arr_double.arr;
+}
+NT_String *NT_GetEntryStringArrFromValue(struct NT_Value *value, unsigned long long *last_change, size_t *arr_size)
+{
+	if (value->type != NT_Type::NT_STRING_ARRAY) return 0;
+	*last_change = value->last_change;
+	*arr_size = value->data.arr_string.size;
+	return value->data.arr_string.arr;
+}
+
+int NT_GetEntryBoolean(const char* name, size_t name_len, unsigned long long *last_change)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	int ret = NT_GetEntryBooleanFromValue(value, last_change);
+	return ret;
+}
+double NT_GetEntryDouble(const char* name, size_t name_len, unsigned long long *last_change)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	double ret = NT_GetEntryDoubleFromValue(value, last_change);
+	return ret;
+}
+struct NT_String NT_GetEntryString(const char *name, size_t name_len, unsigned long long *last_change)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	NT_String ret = NT_GetEntryStringFromValue(value, last_change);
+	return ret;
+}
+struct NT_String NT_GetEntryRaw(const char *name, size_t name_len, unsigned long long *last_change)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	NT_String ret = NT_GetEntryStringFromValue(value, last_change);
+	return ret;
+}
+
+int *NT_GetEntryBooleanArr(const char* name, size_t name_len, unsigned long long *last_change, size_t *arr_size)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	int *ret = NT_GetEntryBooleanArrFromValue(value, last_change, arr_size);
+	return ret;
+}
+double *NT_GetEntryDoubleArr(const char* name, size_t name_len, unsigned long long *last_change, size_t *arr_size)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	double *ret = NT_GetEntryDoubleArrFromValue(value, last_change, arr_size);
+	return ret;
+}
+NT_String *NT_GetEntryStringArr(const char* name, size_t name_len, unsigned long long *last_change, size_t *arr_size)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_GetEntryValue(name, name_len, value);
+	NT_String *ret = NT_GetEntryStringArrFromValue(value, last_change, arr_size);
+	return ret;
+}
```

Comment:
Make sure you have a newline at the end of the file, that's a POSIX standard we should try to conform to.